gms/feature_service.cc

```diff
@@ -284,7 +284,10 @@ future<> feature_service::enable_features_on_startup(db::system_keyspace& sys_ks
 future<> gossiper_feature_enabler::enable_features() {
     auto loaded_peer_features = co_await _sys_ks.load_peer_features();
     auto&& features = _g.get_supported_features(loaded_peer_features, gossiper::ignore_features_of_local_node::no);
+    co_await _pfe.enable(std::move(features));
```

Comment:
Now this looks really weird. The `gossiper_feature_enabler` "knows" that it's persistent and presumes that features-to-enable are stored in system-keyspace, then loads them from system keyspace but to enable-and-persist it relies on external persistent enabler, instead of committing them where it knows they should persist (into sys_ks directly)

Reply:
Well, my intent was that the `gossiper_feature_enabler` is not responsible for "enabling persistently", it delegates this responsibility to the `persistent_feature_enabler`. The fact that persistence is implemented by storing the features in system keyspace is an implementation detail.

That said, I realize more and more that the current implementation is overcomplicated. I was trying to incorporate the approach proposed by Avi here: https://github.com/scylladb/scylladb/pull/13497#discussion_r1212802239 and originally there were multiple types of persisters. However, this is not the right abstraction for cluster features on raft. Cluster features on raft are first persisted in `system.topology` - which is handled by the raft code already - and only after they are persisted we mark them as enabled in the feature service. We no longer treat `system.scylla_local` as the source of truth and do not look at it on startup when checking features - we only want to copy them over there to simulate the current, externally visible behavior (e.g. for the sake of tests, maybe tools).

So, I think I'll drop most of the refactors in the `feature_service` itself and will replace it with the following:

- When applying group 0 commands that modify `system.topology`, call `feature_service::enable` with the current feature set, then write it to `system.scylla_local` as a way of simulating the old behavior.

This will be much simpler and similar to how we currently update `system.peers` based on the group 0 state in `storage_service::topology_state_load`.

Reply:
Done, should be simpler now.